src/recon.rs

```diff
@@ -708,18 +708,22 @@ unsafe fn decode_coefs<BD: BitDepth>(
                 levels[..(stride * (4 * sw as isize + 2)) as usize].fill(0);
                 let mut x: c_uint;
                 let mut y: c_uint;
-                if TX_CLASS_2D as c_int == TX_CLASS_2D as c_int {
-                    rc = *scan.offset(eob as isize) as c_uint;
-                    x = rc >> shift;
-                    y = rc & mask;
-                } else if TX_CLASS_2D as c_int == TX_CLASS_H as c_int {
-                    x = eob as c_uint & mask;
-                    y = (eob >> shift) as c_uint;
-                    rc = eob as c_uint;
-                } else {
-                    x = eob as c_uint & mask;
-                    y = (eob >> shift) as c_uint;
-                    rc = x << shift2 | y;
+                match tx_class {
```

Comment:
this relies on constant propagation to perform this match at compile time, but based on some tests even rust's MIR passes find this now (otherwise LLVM would)